traveler-ios-ui/TravelerKitUI/ViewControllers/PartnerOfferingDetailsViewController.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  PartnerOfferingDetailsViewController.swift
+//  TravelerKitUI
+//
+//  Created by Omar Padierna on 2019-11-27.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+import TravelerKit
+
+public protocol PartnerOfferingDetailsViewControllerDelegate: class {
+    func partnerOfferingDetailsViewController(_ controller: PartnerOfferingDetailsViewController, didFinishWith purchaseForm: PurchaseForm)
+}
+
+open class PartnerOfferingDetailsViewController: UIViewController {
+    @IBOutlet weak var headerView: PagedHorizontalView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var priceLabel: UILabel!
+    @IBOutlet weak var optionsHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var optionsView: UIView!
+
+    var details: PartnerOfferingItemDetail?
+    var product: PartnerOfferingItem?
+    weak var delegate: PartnerOfferingDetailsViewControllerDelegate?
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+
+        titleLabel.text = details?.title
+        descriptionLabel.text = details?.description
+        priceLabel.text = product?.price.localizedDescriptionInBaseCurrency
+
+        // Preload images
+        details?.imageUrls.forEach({ (imageURL) in
```

Comment:
Is this necessary? If you're asyc loading the images as cells render, each invocation should be done at about the same time.

Reply:
...I'm not sure what you mean by "each invocation should be done at the same time". By invoking the asset manager in view did load, you're preemptively making the network request of each image and storing it in a cache. When the asset manager is called in every row, it is now loading images from the cache not necessarily from the network.